shuup/front/views/misc.py

```diff
@@ -7,15 +7,53 @@
 # LICENSE file in the root directory of this source tree.
 from django.http import HttpResponseRedirect
 
-from shuup import configuration
+from shuup.core.models import get_company_contact_for_shop_staff
+from shuup.core.utils.users import (
+    force_anonymous_contact_for_user, force_person_contact_for_user,
+    toggle_all_seeing_for_user
+)
 from shuup.front.utils.user import is_admin_user
 
 
 def toggle_all_seeing(request):
     return_url = request.META["HTTP_REFERER"]
     if not is_admin_user(request):
         return HttpResponseRedirect(return_url)
-    all_seeing_key = "is_all_seeing:%d" % request.user.pk
-    is_all_seeing = not configuration.get(None, all_seeing_key, False)
-    configuration.set(None, all_seeing_key, is_all_seeing)
+
+    toggle_all_seeing_for_user(request.user)
+    return HttpResponseRedirect(return_url)
+
+
+def force_anonymous_contact(request):
```

Comment:
I guess you are repeating yourself with this logic. I guess you could create a decorator for these `is_admin_user` checks

Reply:
Yeah likely, but is it worth. Likely not. And I guess it is not guaranteed that we want same logic to all of these.